spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -270,223 +315,465 @@ def __call__(self, _, namespace, values, option_string=None):
+        hdr=current_value.hdr,
+    )
 
-        printv('\nDone! File created: ' + fname_out, verbose, 'info')
 
-    # Case 2: One or more array operations (array_in -> array_out)
+def apply_otsu_median(current_value, size, n_iter):
+    """Implementation for -otsu-median."""
+    return Image(
+        sct_math.otsu_media(current_value.data, size, n_iter),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_percent(current_value, percentile):
+    """Implementation for -percent."""
+    return Image(
+        sct_math.perc(current_value.data, percentile),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_thr(current_value, threshold):
+    """Implementation for -thr."""
+    return Image(
+        sct_math.threshold(current_value.data, lthr=threshold),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_uthr(current_value, threshold):
+    """Implementation for -uthr."""
+    return Image(
+        sct_math.threshold(current_value.data, uthr=threshold),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_dilate(current_value, size, *, shape, dim):
+    """Implementation for -dilate."""
+    return Image(
+        sct_math.dilate(current_value.data, size=size, shape=shape, dim=dim),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_erode(current_value, size, *, shape, dim):
+    """Implementation for -erode."""
+    return Image(
+        sct_math.erode(current_value.data, size=size, shape=shape, dim=dim),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_smooth(current_value, *sigmas):
+    """Implementation for -smooth."""
+    ndims = current_value.data.ndims
+    if len(sigmas) == 1:
+        sigmas *= ndims
+    elif len(sigmas) != ndims:
+        raise SctMathsValueError(f"expected 1 or {ndims} values, got {len(sigmas)} values")
+
+    # adjust sigma based on voxel size
+    pdims = current_value.dim[4:8]
+    sigmas = [s/d for s, d in zip(sigmas, pdims)]
+
+    return Image(
+        sct_math.smooth(current_value.data, sigmas),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_laplacian(current_value, *sigmas):
+    """Implementation for -laplacian."""
+    ndims = current_value.data.ndims
+    if len(sigmas) == 1:
+        sigmas *= ndims
+    elif len(sigmas) != ndims:
+        raise SctMathsValueError(f"expected 1 or {ndims} values, got {len(sigmas)} values")
+
+    # adjust sigma based on voxel size
+    pdims = current_value.dim[4:8]
+    sigmas = [s/d for s, d in zip(sigmas, pdims)]
+
+    return Image(
+        sct_math.laplacian(current_value.data, sigmas),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_denoise(current_value, patch_radius, block_radius):
+    """Implementation for -denoise."""
+    return Image(
+        sct_math.denoise_nlmeans(current_value.data, patch_radius, block_radius),
+        hdr=current_value.hdr,
+    )
+
+
+def apply_mi(current_value, fname):
+    """Implementation for -mi."""
+    return (current_value, Image(fname), 'mi', 'Mutual information')
```

Comment:
I'm still on-board for turning the similarity metrics into their own SCT command outside of `sct_maths`, but I think this design is livable until then.